webclient/src/websocket/commands/session/register.ts

```diff
@@ -0,0 +1,78 @@
+import { ServerRegisterParams } from "store";
+import { WebSocketConnectOptions } from "types";
+
+import webClient from "../../WebClient";
+import { SessionPersistence } from "../../persistence";
+import { hashPassword } from "../../utils";
+import NormalizeService from "../../utils/NormalizeService";
+
+import { login, disconnect } from "./";
+
+export function register(options: WebSocketConnectOptions, passwordSalt?: string): void {
+  const { userName, password, email, country, realName } = options as ServerRegisterParams;
+
+  const registerConfig: any = {
+    ...webClient.clientConfig,
+    userName,
+    email,
+    country,
+    realName,
+  };
+
+  if (passwordSalt) {
+    registerConfig.hashedPassword = hashPassword(passwordSalt, password);
+  } else {
+    registerConfig.password = password;
+  }
+
+  const CmdRegister = webClient.protobuf.controller.Command_Register.create(registerConfig);
+
+  const sc = webClient.protobuf.controller.SessionCommand.create({
+    '.Command_Register.ext': CmdRegister
+  });
+
+  webClient.protobuf.sendSessionCommand(sc, raw => {
+    if (raw.responseCode === webClient.protobuf.controller.Response.ResponseCode.RespRegistrationAccepted) {
```

Comment:
Any reason we call out this one case first here? Could we just combine it into the switch below?